server/src/main/java/org/opensearch/index/mapper/TextFieldMapper.java

```diff
@@ -1168,24 +1193,25 @@ public static Query createPhrasePrefixQuery(
     protected void doXContentBody(XContentBuilder builder, boolean includeDefaults, Params params) throws IOException {
         // this is a pain, but we have to do this to maintain BWC
         builder.field("type", contentType());
-        this.builder.boost.toXContent(builder, includeDefaults);
-        this.builder.index.toXContent(builder, includeDefaults);
-        this.builder.store.toXContent(builder, includeDefaults);
+        Builder mapperBuilder = (TextFieldMapper.Builder) getMergeBuilder();
```

Comment:
Earlier, the builder object was in-memory but now it will initiate a new object on demand but that would be acceptable compared to the memory gain we will get by not keeping builder around all the time.

Reply:
I wonder why do we need to use builder object at all can't we simply serialize each field param explicitly

Reply:
Once the `TextFieldMapper` object is created the `Parameter` objects are parsed stored differently within the object as separate data types like boolean, string, number etc. In order to serialize, we need to construct the`Parameter` objects back from the data within the `TextFieldMapper` instance because the `Parameter` class has the [logic](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/mapper/ParametrizedFieldMapper.java#L343-L344) to decide whether a particular field param has to be serialized or not.

Reply:
For now, lets go ahead with the current improvement of removing builder. I want to evaluate if we should simplify the serialization logic further